src/main/java/de/tum/in/www1/artemis/service/hestia/CodeHintService.java

```diff
@@ -0,0 +1,63 @@
+package de.tum.in.www1.artemis.service.hestia;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.hestia.CodeHint;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.hestia.CodeHintRepository;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * Service for managing code hints.
+ */
+@Service
+public class CodeHintService {
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final CodeHintRepository codeHintRepository;
+
+    public CodeHintService(ProgrammingExerciseRepository programmingExerciseRepository, CodeHintRepository codeHintRepository) {
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.codeHintRepository = codeHintRepository;
+    }
+
+    /**
+     * Create a codeHint
+     * @param codeHint
+     * @return created code hint
+     */
+    public CodeHint createCodeHint(CodeHint codeHint) {
```

Comment:
delete this method, you are adding error handling that is already provided by the underlying Hibernate implementation